flink-runtime/src/test/java/org/apache/flink/runtime/rest/MultipartUploadExtension.java

```diff
@@ -119,6 +120,12 @@ public void before(ExtensionContext context) throws Exception {
         }
         file2 = TempDirUtils.newFile(tmpDirectory);
         Files.write(file2.toPath(), "world".getBytes(ConfigConstants.DEFAULT_CHARSET));
+        file3 = TempDirUtils.newFile(tmpDirectory);
+        try (RandomAccessFile rw = new RandomAccessFile(file3, "rw")) {
+            // magic value that reliably reproduced EndOfDataDecoderException in hasNext()
+            rw.setLength(1410);
```

Comment:
The test doesn't fail if I revert the fix. That makes me believe that this magic value doesn't reproduce the scenario reliably. How did you come up with the test scenario? :thinking:

Reply:
It's not 100% reliable, but it worked for me pretty well.

Just tried with different request sizes.

Reply:
meh now it stopped failing for me, hold on. Could be that some of the refactorings changed the number.

Reply:
@XComp can you try again with the new magic number?

Reply:
These things are very susceptible to interference from other requests or parts of the payload; for example if the JSON payload increases by a single byte it can already break the number.

Reply:
If it's worth anything, the magic number break things for me 🪄 

```
[ERROR] Tests run: 9, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 3.23 s <<< FAILURE! - in org.apache.flink.runtime.rest.FileUploadHandlerITCase
[ERROR] org.apache.flink.runtime.rest.FileUploadHandlerITCase.testMixedMultipartEndOfDataDecoderExceptionHandling(Path)  Time elapsed: 0.099 s  <<< FAILURE!
org.opentest4j.AssertionFailedError:

expected: 200
 but was: 500
	at java.base/jdk.i
```

Reply:
Ya, it fails for me now quite reliably as well with the fix being commented-out.

> These things are very susceptible to interference from other requests or parts of the payload; for example if the JSON payload increases by a single byte it can already break the number.

Is it worth adding this information to the test method?

Reply:
done